sycl/include/sycl/ext/oneapi/matrix/matrix-hip.hpp

```diff
@@ -0,0 +1,452 @@
+  template <sycl::ext::oneapi::experimental::matrix::layout Layout>            \
+  struct joint_matrix_hip<                                                     \
+      TYPE, sycl::ext::oneapi::experimental::matrix::use::USE, M, N, Layout,   \
+      typename std::enable_if_t<                                               \
+          Layout ==                                                            \
+              sycl::ext::oneapi::experimental::matrix::layout::row_major ||    \
+          Layout ==                                                            \
+              sycl::ext::oneapi::experimental::matrix::layout::col_major>> {   \
+    sycl::marray<TYPE, SIZE> data;                                             \
+  };
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, a, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, b, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, a, 32, 8, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, b, 8, 32, 4)
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, a, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, b, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, a, 32, 8, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, b, 8, 32, 4)
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(double, a, 16, 4, 1)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(double, b, 4, 16, 1)
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(int8_t, a, 32, 8, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(int8_t, b, 8, 32, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(int8_t, a, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(int8_t, b, 16, 16, 4)
+
+#undef __SYCL_JOINT_MATRIX_OVERLOAD_ARR
+
+#define __SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(TYPE, M, N)                       \
+  template <>                                                                  \
+  struct joint_matrix_hip<                                                     \
+      TYPE, sycl::ext::oneapi::experimental::matrix::use::accumulator, M, N,   \
+      sycl::ext::oneapi::experimental::matrix::layout::dynamic> {              \
+    sycl::marray<TYPE, (M * N) / 64> data;                                     \
+  };
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(float, 16, 16)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(float, 32, 32)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(double, 16, 16)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(int32_t, 32, 32)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(int32_t, 16, 16)
+
+#undef __SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC
+
+template <sycl::ext::oneapi::experimental::matrix::layout Layout, typename S,
+          typename T, size_t M, size_t N, access::address_space Space,
+          access::decorated IsDecorated, typename Group>
+void load_accumulator_layoutT(
+    joint_matrix_hip<
+        S, sycl::ext::oneapi::experimental::matrix::use::accumulator, M, N,
+        sycl::ext::oneapi::experimental::matrix::layout::dynamic> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, Group &sg) {
+  const auto idx = sg.get_group_linear_id() * sg.get_local_range()[0] +
```

Comment:
@JackAKirk do you understand here why the AMX backend needs to query ids using the runtime sg variable while this was not needed in the CUDA backend?

Reply:
@dkhaldi The id is associated to 64 lanes of wavefront. Matrix data should be loaded to lane's data before mad.
In cuda there are instructions to load / store matrix data before mad.

Reply:
Thanks @mmoadeli, I think I see.
The Hip backend uses WI level assignments (a[i] = A[a_idx];) to load data from memory to vectors in registers.
These assignments depend on the position of the WI in the warp. 
That's why you need to pass the sg argument.
Is my understanding correct?

Reply:
That's right @dkhaldi